sycl/test-e2e/Matrix/joint_matrix_colA_rowB_colC_impl.hpp

```diff
@@ -0,0 +1,130 @@
+           // subgroup these functions will be called once by the subgroup no
+           // code divergence between the workitems
+           const auto global_idx = spmd_item.get_global_id(0);
+           const auto global_idy = spmd_item.get_global_id(1);
+           const auto sg_startx = global_idx - spmd_item.get_local_id(0);
+           const auto sg_starty = global_idy - spmd_item.get_local_id(1);
+
+           sub_group sg = spmd_item.get_sub_group();
+           joint_matrix<sub_group, bfloat16, use::a, TM, TK, layout::col_major>
+               sub_a;
+           joint_matrix<sub_group, bfloat16, use::b, TK, TN, layout::row_major>
+               sub_b;
+           joint_matrix<sub_group, float, use::accumulator, TM, TN> sub_c;
+           joint_matrix_fill(sg, sub_c, 1);
+           for (int k = 0; k < K; k += TK) {
+             joint_matrix_load(sg, sub_a, pA + (sg_startx * TM) * K + k, K);
+             joint_matrix_load(
+                 sg, sub_b, pB + (k) * (N) + sg_starty / SG_SZ * TN * 2, N * 2);
+             sub_c = joint_matrix_mad(sg, sub_a, sub_b, sub_c);
+           }
+           joint_matrix_store(
+               sg, sub_c, pC + (sg_startx * TM) * N + sg_starty / SG_SZ * TN, N,
+               layout::col_major);
+         }); // parallel for
+   }).wait();
+}
+
+void matrix_multiply_ref(bfloat16 *A, bfloat16 *B, float *C, int MATRIX_M,
+                         int MATRIX_N, int MATRIX_K) {
+  for (unsigned int i = 0; i < MATRIX_M; i++) {
+    for (unsigned int k = 0; k < MATRIX_K; k++) {
+      for (unsigned int j = 0; j < MATRIX_N; j++) {
+        C[j * MATRIX_N + i] +=
+            make_fp32(A[i * MATRIX_K + k]) * make_fp32(B[k * MATRIX_N + j]);
+      }
+    }
+  }
+}
+
+int main() {
+  static constexpr size_t MATRIX_M = 1024;
+  static constexpr size_t MATRIX_N = 1024;
+  static constexpr size_t MATRIX_K = 1024;
+  queue q;
+  bfloat16 *A = malloc_shared<bfloat16>(MATRIX_M * MATRIX_K, q);
+  bfloat16 *B = malloc_shared<bfloat16>(MATRIX_K * MATRIX_N, q);
+  float *C = malloc_shared<float>(MATRIX_M * MATRIX_N, q);
+  float *D = malloc_shared<float>(MATRIX_M * MATRIX_N, q);
+
+  std::random_device dev;
+  std::uniform_real_distribution<float> fdistr(-1.0, 1.0);
+
+  for (int i = 0; i < MATRIX_M; i++) {
+    for (int j = 0; j < MATRIX_K; j++) {
+      A[i * MATRIX_K + j] = bfloat16(fdistr(dev));
+    }
+  }
+  for (int i = 0; i < MATRIX_K; i++) {
+    for (int j = 0; j < MATRIX_N; j++) {
+      B[i * MATRIX_N + j] = bfloat16(fdistr(dev));
+    }
+  }
+  for (int i = 0; i < MATRIX_M; i++) {
+    for (int j = 0; j < MATRIX_N; j++) {
+      C[i * MATRIX_N + j] = 1.0;
+      D[i * MATRIX_N + j] = 1.0;
+    }
+  }
+
+  matrix_multiply<float, bfloat16, MATRIX_M, MATRIX_K, MATRIX_K, MATRIX_N,
+                  MATRIX_M, MATRIX_N>(C, A, B, q);
+  matrix_multiply_ref(A, B, D, MATRIX_M, MATRIX_N, MATRIX_K);
+
+  bool res = true;
+  for (int i = 0; i < MATRIX_M; i++) {
```

Comment:
I noticed verification code is the same across almost all tests. Please move it as a function to common.hpp as well.

Reply:
please, use matrix_compare from common.hpp